### ISSUE TITLE
Node.js  Loaders Team Meeting 2023-05-23

### ISSUE THREAD

**Issue by mhdawson** [nodejs/loaders#141]
## Time

**UTC Tue 23-May-2023 18:00 (06:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Tue 23-May-2023 11:00 (11:00 AM) |
| US / Mountain | Tue 23-May-2023 12:00 (12:00 PM) |
| US / Central  | Tue 23-May-2023 13:00 (01:00 PM) |
| US / Eastern  | Tue 23-May-2023 14:00 (02:00 PM) |
| EU / Western  | Tue 23-May-2023 19:00 (07:00 PM) |
| EU / Central  | Tue 23-May-2023 20:00 (08:00 PM) |
| EU / Eastern  | Tue 23-May-2023 21:00 (09:00 PM) |
| Moscow        | Tue 23-May-2023 21:00 (09:00 PM) |
| Chennai       | Tue 23-May-2023 23:30 (11:30 PM) |
| Hangzhou      | Wed 24-May-2023 02:00 (02:00 AM) |
| Tokyo         | Wed 24-May-2023 03:00 (03:00 AM) |
| Sydney        | Wed 24-May-2023 04:00 (04:00 AM) |

Or in your local time:
* https://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Loaders%20Team+Meeting+2023-05-23&iso=20230523T1800
* or https://www.wolframalpha.com/input/?i=06PM+UTC%2C+May+23%2C+2023+in+local+time

## Links

* Minutes Google Doc: <https://docs.google.com/document/d/1ZbImY9nugbcEQEnUefUOW0aKdioAeqkT-q8PbcuDaW8/edit>

## Agenda

Extracted from **loaders-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/node

* console.log failure while working with worker threads. [#30491](https://github.com/nodejs/node/issues/30491)

## Invited

* Loaders team: @nodejs/loaders

## Notes

The agenda comes from issues labelled with `loaders-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

* link for participants: <https://zoom.us/j/97506450082>

* For those who just want to watch: <https://www.youtube.com/c/nodejs+foundation/live>

**Comment by JakobJingleheimer** [comment]
I think we have nothing to discuss? (I'm available if we do)

**Comment by aduh95** [comment]
There are two things that come to mind:

- how to make `registerLoader` happen (what the final API should look like, and what are the steps (ideally each step should be as small as possible, so we can delegate some of the work to e.g. first-time contributors looking for ideas on how to contribute)
- how to make CJS loading from the ESM loader happen. I have a draft PR open which is at the POC stage, and I'm not going to complete the work by myself, so unless we're able to dispatch the work async, we could use the meeting time to discuss that. In any case, it would certainly be helpful if everyone interested could pull the PR locally and play with it (and bonus point if someone wants to contribute to testing all the crazy edge cases we'd need to have in mind).

**Comment by aduh95** [comment]
Yes, my PR goal is to run the code is a CJS-like environment when the `load` step returns `{ format: 'commonjs', source: … }`.

**Comment by isaacs** [comment]
> @isaacs have you started working on that?

I haven't started in earnest yet, but have been reading code and it's bubbling up my list swiftly.  If @aduh95 or anyone else wants to get started, don't let me hold you back.

**Comment by GeoffreyBooth** [comment]
> @aduh95 Can you share a link to the draft PR? I should have some time to poke at it in advance of the meeting.

https://github.com/nodejs/node/pull/47999

**Comment by JakobJingleheimer** [comment]
I think `registerLoader` is done in principle (there are just some test cases needed, which I'll get to in a few days if João doesn't get to it first).
